patches/minecraft/net/minecraft/entity/Entity.java.patch

```diff
@@ -240,7 +245,41 @@
  
              double d3 = Math.min(-2.9999872E7D, serverworld1.func_175723_af().func_177726_b() + 16.0D);
              double d4 = Math.min(-2.9999872E7D, serverworld1.func_175723_af().func_177736_c() + 16.0D);
-@@ -2118,7 +2141,7 @@
+@@ -2099,26 +2127,32 @@
+             d1 = MathHelper.func_151237_a(d1, d4, d6);
+             Vec3d vec3d1 = this.func_181014_aG();
+             blockpos = new BlockPos(d0, this.field_70163_u, d1);
+-            BlockPattern.PortalInfo blockpattern$portalinfo = serverworld1.func_85176_s().func_222272_a(blockpos, vec3d, this.func_181012_aH(), vec3d1.field_72450_a, vec3d1.field_72448_b, this instanceof PlayerEntity);
+-            if (blockpattern$portalinfo == null) {
+-               return null;
+-            }
++            if (teleporter.isVanilla()) {
++               BlockPattern.PortalInfo blockpattern$portalinfo = ((net.minecraft.world.Teleporter) teleporter).func_222272_a(blockpos, vec3d, this.func_181012_aH(), vec3d1.field_72450_a, vec3d1.field_72448_b, this instanceof PlayerEntity);
++               if (blockpattern$portalinfo == null) {
++                  return null;
++               }
+ 
+-            blockpos = new BlockPos(blockpattern$portalinfo.field_222505_a);
+-            vec3d = blockpattern$portalinfo.field_222506_b;
+-            f = (float)blockpattern$portalinfo.field_222507_c;
++               blockpos = new BlockPos(blockpattern$portalinfo.field_222505_a);
++               vec3d = blockpattern$portalinfo.field_222506_b;
++               f = (float) blockpattern$portalinfo.field_222507_c;
++            } else {
++               teleporter.placeEntity(this, serverworld, serverworld1, this.field_70177_z);
```

Comment:
In vanilla, this boolean value is used to decide whether a new nether portal has to be built. I think using this same return value to prevent teleports or generate a portal depending on whether it's a vanilla teleporter wouldn't be very useful. Teleporting can already be prevented by canceling the EntityTravelToDimensionEvent so I think it would also be a little bit redundant.